src/test/java/teammates/sqlui/webapi/UpdateFeedbackSessionLogsActionTest.java

```diff
@@ -0,0 +1,207 @@
+        mockLogsProcessor.insertFeedbackSessionLog(student1, feedbackSession1,
+                FeedbackSessionLogType.SUBMISSION.getLabel(),
+                startTime.plusSeconds(300).toEpochMilli());
+        mockLogsProcessor.insertFeedbackSessionLog(student1, feedbackSession1,
+                FeedbackSessionLogType.VIEW_RESULT.getLabel(),
+                startTime.plusSeconds(300).toEpochMilli());
+
+        // Different feedback sessions
+        mockLogsProcessor.insertFeedbackSessionLog(student1, feedbackSession1,
+                FeedbackSessionLogType.ACCESS.getLabel(),
+                startTime.plusSeconds(600).toEpochMilli());
+        mockLogsProcessor.insertFeedbackSessionLog(student1, feedbackSession2,
+                FeedbackSessionLogType.ACCESS.getLabel(),
+                startTime.plusSeconds(600).toEpochMilli());
+
+        // Different Student
+        mockLogsProcessor.insertFeedbackSessionLog(student1, feedbackSession1,
+                FeedbackSessionLogType.ACCESS.getLabel(),
+                startTime.plusSeconds(900).toEpochMilli());
+        mockLogsProcessor.insertFeedbackSessionLog(student2, feedbackSession1,
+                FeedbackSessionLogType.ACCESS.getLabel(),
+                startTime.plusSeconds(900).toEpochMilli());
+
+        // Gap is larger than spam filter
+        mockLogsProcessor.insertFeedbackSessionLog(student1, feedbackSession1,
+                FeedbackSessionLogType.ACCESS.getLabel(), startTime.toEpochMilli());
+        mockLogsProcessor.insertFeedbackSessionLog(student1, feedbackSession1,
+                FeedbackSessionLogType.ACCESS.getLabel(),
+                startTime.plusMillis(SPAM_FILTER + 1).toEpochMilli());
+
+        UpdateFeedbackSessionLogsAction action = getAction();
+        action.execute();
+
+        // method returns all logs regardless of params
+        List<FeedbackSessionLogEntry> expected =
+                mockLogsProcessor.getOrderedFeedbackSessionLogs("", "", 0, 0, "");
+
+        verify(mockLogic).createFeedbackSessionLogs(
+                argThat(filteredLogs -> isEqualExceptId(expected, filteredLogs)));
+    }
+
+    @Test
+    public void testExecute_recentLogsWithSpam_someLogsCreated()
+            throws EntityAlreadyExistsException, InvalidParametersException {
+        // Gap is smaller than spam filter
+        mockLogsProcessor.insertFeedbackSessionLog(student1, feedbackSession1,
+                FeedbackSessionLogType.ACCESS.getLabel(), startTime.toEpochMilli());
+        mockLogsProcessor.insertFeedbackSessionLog(student1, feedbackSession1,
+                FeedbackSessionLogType.ACCESS.getLabel(),
+                startTime.plusMillis(SPAM_FILTER - 2).toEpochMilli());
+
+        // Filters multiple logs within one spam window
+        mockLogsProcessor.insertFeedbackSessionLog(student1, feedbackSession1,
+                FeedbackSessionLogType.ACCESS.getLabel(),
+                startTime.plusMillis(SPAM_FILTER - 1).toEpochMilli());
```

Comment:
Just a very small nit, not important, can order them ascending (in the order they'll be processed)

Reply:
Sorry I don't quite get the comment is this not in ascending order?